setup.sh

```diff
@@ -30,7 +30,24 @@ function usage
 	exit 1
 }
 
+# We must do this check before the `declare`, because MacOS ships with bash version 3
+[ "$(uname)" == "Darwin" ] && IS_MACOS=1 || IS_MACOS=0
+
+if ((BASH_VERSINFO[0] < 4));
+then
+	echo "This script requires bash version >= 4.0, and you have bash verison $BASH_VERSION." >&2
+	if [ $IS_MACOS -eq 1 ];
+	then
+		echo -e "To install a newer bash version, use homebrew https://brew.sh/:\nbrew install bash\nYou don't need to link it or change the shell, it just needs to be installed." >&2
```

Comment:
Should we just do this automatically if the -i flat is present?

Reply:
Probably, and I think I could get that to work, but I'd need to do a similar `homebrew` check here.

Perhaps I'll put the homebrew check into a function so that I can call it in both places.

Reply:
Actually, on second look, we can't put this check after the `declare`
https://github.com/angr/angr-dev/blob/master/setup.sh#L37

So we'd have to parse the `-i` flag earlier in the script. Probably not worth it IMHO.